chapters/classes.tex

```diff
@@ -351,62 +373,98 @@ \subsection{Acyclic Bindings of Constants and Parameters}\label{acyclic-bindings
 \end{lstlisting}
 \end{example}
 
-\subsection{Component Variability Prefixes}\label{component-variability-prefixes-discrete-parameter-constant}
 
-The prefixes \lstinline!discrete!, \lstinline!parameter!, \lstinline!constant! of a component declaration are called \firstuse[variability!prefix]{variability prefixes}\index{component variability}\index{declared variability}\index{variability!declared|see{declared variability}} and define in which situation the variable values of a component are initialized (see \cref{events-and-synchronization} and \cref{initialization-initial-equation-and-initial-algorithm}) and when they are changed during simulation:
+\subsection{Conditional Component Declaration}\label{conditional-component-declaration}
+
+A component declaration can have a \lstinline!condition-attribute!: \lstinline!if!~\emph{expression}.
+
+\begin{example}
+\begin{lstlisting}[language=modelica]
+  parameter Integer level(min=1)=1;
+  Motor motor;
+  Level1 component1(J=J) if level==1 "Conditional component";
+  Level2 component2 if level==2 "Conditional component";
+  Level3 component3(J=component1.J) if level<2 "Conditional component";
+  // Illegal modifier on component3 since component1.J is conditional
+  // Even if we can see that component1 always exist if component3 exist
+equation
+  connect(component1$\ldots$, $\ldots$) "Connection to conditional component 1";
+  connect(component2.n, motor.n) "Connection to conditional component 2";
+  connect(component3.n, motor.n) "Connection to conditional component 3";
+  component1.u=0; // Illegal
+\end{lstlisting}
+\end{example}
+
+The \emph{expression} must be a \lstinline!Boolean! scalar expression, and must be a parameter expression.
+
+\begin{nonnormative}
+A parameter expression is required since it shall be evaluated at compile time.
```

Comment:
```suggestion
An evaluable expression is required since it shall be evaluated at compile time.
```

Reply:
Should to into same PR as the suggested change above.

Reply:
As it's a non-normative comment to the normative change that should go into a separate PR, it seems best to work out the two in conjunction.  When we get to do that, it's worth reconsidering the strong wording with _shall_ used inside of a non-normative section.  Maybe it's better to let the evaluable variability speak for itself, and not arbitrarily require that some evaluable expressions must be evaluated during translation…

We also have a similar strong formulation to look at for the `assert`-equation:
```
\begin{nonnormative}
A parameter expression is required for \lstinline!level! since it shall be evaluated at compile time.
\end{nonnormative}
```

Reply:
I take the approving review as a sign that we continue this discussion in a new PR.  Resolving.